packages/datagrid/src/components/DatasetSerializer/datasetSerializer.js

```diff
@@ -67,10 +85,7 @@ export function getTypeValue(type, optional) {
  */
 export function getType(type) {
 	if (Array.isArray(type)) {
-		return getTypeValue(
-			type.find(subType => subType.type !== 'null'),
-			type.find(subType => subType.type === 'null'),
-		);
+		return getTypeValue(type.find(negate(isNull)), type.find(isNull));
```

Comment:
This seems odd...
Can you explain ?

Reply:
Yep I had the same feeling.
From what I understand : 
- take the type label of the first non null type
- it is mandatory if there is no type that is "null"

Perhaps you can name functions better
```javascript
return getTypeValue(getFirstNonNull(type), canBeOptional(type));
```

But is it ok to get the label of only the first type ?